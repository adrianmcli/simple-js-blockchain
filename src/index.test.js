const { genesisBlock, blockAddTx, hashBlock, mineBlock } = require('./index')

test('blockAddTx', () => {
  const newTx = { from: "Alice", to: "Bob", amount: 10 }
  expect(blockAddTx(genesisBlock, newTx)).toMatchSnapshot()
})

describe('hashBlock', () => {
  it('should hash the genesisBlock', () => {
    expect(hashBlock(genesisBlock)).toMatchSnapshot()
  })

  it('should hash a non-genesis block', () => {
    const newTx = { from: "Alice", to: "Bob", amount: 10 }
    const nonGenesisBlock = blockAddTx(genesisBlock, newTx)
    expect(hashBlock(nonGenesisBlock)).toMatchSnapshot()
  })
})

describe('mineBlock', () => {
  it('should mine the genesisBlock', () => {
    expect(mineBlock(genesisBlock, 0)).toMatchSnapshot()
  })

  it('should mine a non-genesis block', () => {
    const newTx = { from: "Alice", to: "Bob", amount: 10 }
    const nonGenesisBlock = blockAddTx(genesisBlock, newTx)
    expect(mineBlock(nonGenesisBlock, 0)).toMatchSnapshot()
  })
})
