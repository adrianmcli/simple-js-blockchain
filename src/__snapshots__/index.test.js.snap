// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blockAddTx 1`] = `
Object {
  "hash": "",
  "index": 0,
  "nonce": null,
  "prevHash": "000000000000000000000000000000000",
  "txs": Array [
    Object {
      "amount": 20,
      "from": "Heaven",
      "to": "Adrian",
    },
    Object {
      "amount": 10,
      "from": "Alice",
      "to": "Bob",
    },
  ],
}
`;

exports[`hashBlock should hash a non-genesis block 1`] = `"0c3f2782a3d5cfc2c7a868b2afd858ac304134283f83f1eb7466cd57a107cfaa"`;

exports[`hashBlock should hash the genesisBlock 1`] = `"978bcbca07e6258e085f4d2bfcc8a332fd01c638db96f18ba267893ff95eb91a"`;

exports[`mineBlock should mine a non-genesis block 1`] = `
Object {
  "hash": "0c3f2782a3d5cfc2c7a868b2afd858ac304134283f83f1eb7466cd57a107cfaa",
  "index": 0,
  "nonce": 2,
  "prevHash": "000000000000000000000000000000000",
  "txs": Array [
    Object {
      "amount": 20,
      "from": "Heaven",
      "to": "Adrian",
    },
    Object {
      "amount": 10,
      "from": "Alice",
      "to": "Bob",
    },
  ],
}
`;

exports[`mineBlock should mine the genesisBlock 1`] = `
Object {
  "hash": "978bcbca07e6258e085f4d2bfcc8a332fd01c638db96f18ba267893ff95eb91a",
  "index": 0,
  "nonce": 57,
  "prevHash": "000000000000000000000000000000000",
  "txs": Array [
    Object {
      "amount": 20,
      "from": "Heaven",
      "to": "Adrian",
    },
  ],
}
`;
